vignettes/advanced.Rmd

```diff
@@ -0,0 +1,391 @@
+that should be applied to each post-ICE visit, whilst
+`dlag` specifies the scaling coefficient to be applied based upon the visits proximity
+to first visit affected by the ICE. By default, the delta will only be added to unobserved (i.e. imputed) post-ICE 
+outcomes but this can be changed by setting the optional argument `missing_only = FALSE`.
+
+The usage of the `delta` and `dlag` arguments is best illustrated with a few examples:
+
+Let `delta = c(5,6,7,8)` and `dlag=c(1,2,3,4)` (i.e. assuming there are 4 visits) and lets 
+say that the subject's first visit affected by the ICE is visit 2. The calculation would then be as follows:
+
+```
+v1  v2  v3  v4
+--------------
+ 5   6   7   8  # delta assigned to each visit
+ 0   1   2   3  # scaling starting from the first visit after the subjects ICE
+--------------
+ 0   6  14  24  # delta * scaling
+--------------
+ 0   6  20  44  # accumulative sum / delta to be applied to each visit
+```
+
+That is to say the subject would have a delta offset of 0 applied for visit v1, 6 for 
+visit v2, 20 for visit v3 and 44 for visit v4. As a comparison, lets say that the subject 
+instead that the subject's first visit affected by the ICE was visit 3. Then, would be as follows:
+
+```
+v1  v2  v3  v4
+--------------
+ 5   6   7   8  # delta assigned to each visit
+ 0   0   1   2  # scaling starting from the first visit after the subjects ICE
+--------------
+ 0   0   7  16  # delta * scaling
+--------------
+ 0   0   7  23  # accumulative sum / delta to be applied to each visit
+ ```
+ 
+If one alternatively wanted to apply a constant delta value of 5 to all visits affected by the ICE
+regarrdless of their proximity to the first ICE visit, this could be achieved
+by setting `delta = c(5,5,5,5)` and `dlag = c(1,0,0,0)`. For example lets say the first visit v1 
+was already affected by the ICE for a subject. Then the calculation would be as follows:
+
+```
+v1  v2  v3  v4
+--------------
+ 5   5   5   5  # delta assigned to each visit
+ 1   0   0   0  # scaling starting from the first visit after the subjects ICE
+--------------
+ 5   0   0  0  # delta * scaling
+--------------
+ 5   5   5  5  # accumulative sum / delta to be applied to each visit
+ ```
+
+!!! The example below used the wording "the ICE occurred at" but in data_ICE we only know the first visit affected by the ICE.
+!!! I tried to clean up the wording accordingly but ran into some problems. Please help to fix this! 
+
+Another way of using these arguments is to set delta to be the difference in time 
+between visits and dlag to be the amount of delta per unit of time. For example 
+lets say that we have a visit on weeks 1, 5, 6 & 9 and that we want a delta of 3
+to be applied for each week after an ICE. This can be achieved by setting 
+`delta = c(0,4,1,3)` (the difference in weeks between each visit) and `dlag = c(3, 3, 3, 3)`. 
+For example lets say we have a subject whose first visit affected by the ICE was week 5 (i.e. visit v2) then 
```

Comment:
```suggestion
For example lets say we have a subject who experiences an ICE between visit v1 and v2. Lets assume in particular that the ICE happens at half way between visit v1 and v2. Then the first fisit affected by the ICE would be visit v2 (i.e at week 5) and
```

Reply:
Only issue is that the dlag delta is a blanket rule applied to all subjects so the way this is written invokes the question of "but what if I know that subject A has theirs on week 2 but subject B has theirs on week 4"